src/Adapter/Product/AdminProductWrapper.php

```diff
@@ -496,19 +496,48 @@ public function processProductCustomization($product, $data)
 
         $shopList = Shop::getContextListShopID();
 
+        /** Update the customization fields to be deleted in the next step if not used */
+        $updateQuery = 'UPDATE `' . _DB_PREFIX_ . 'customization_field` cf
+            SET cf.`is_deleted` = 1
+            WHERE
+            cf.`id_product` = ' . (int)$product->id . ' 
```

Comment:
- Did you check this product id before using it in an update query ? (update permissions, product linked to the good shop etc. I'm just throwing ideas of things that could need to be checked.)
- Please do not build and/or execute SQL queries here. Adapters are not model classes.